docs/going-into-production.rst

```diff
@@ -0,0 +1,198 @@
+
+Switching to a manual bootstrapping configuration is the first step towards
+going into production. The rest of this how-to guide covers additional things
+you may want to consider.
+
+
+.. _prod-config:
+
+Additional considerations
+=========================
+
+The following setting can be changed in your `configuration`_ file or set via
+command-line arguments. Additionally, you can change some settings at
+`runtime`_.
+
+
+.. _prod-config-path:
+
+Paths
+-----
+
+In a production environment, you should change the default locations of the
+data and log folders to avoid accidental deletion.
+
+Path configuration is especially important if you are running CrateDB on
+Docker. Any data you want to persist between container restarts should be
+located on a mounted volume.
+
+.. SEEALSO::
+
+    `Path settings`_
+
+
+.. _prod-config-cluster-name:
+
+Cluster name
+------------
+
+The `cluster.name`_ setting allows you to create multiple separate clusters. A
+node will refuse to join a cluster if the respective cluster names do not
+match.
+
+If you are running multiple clusters on the same network, you should set unique
+cluster names.
+
+.. SEEALSO::
+
+    :ref:`Cluster names for multi-node setups <multi-node-cluster-name>`
+
+
+.. _prod-config-node-name:
+
+Node name
+---------
+
+By default, CrateDB sets the node name for you. However, if you configure the
+node names explicitly, you can specify a list of master-eligible nodes
+up-front.
+
+You can configure a custom node name using the `node.name`_ setting.
+
+.. SEEALSO::
+
+    :ref:`Node names for multi-node setups <multi-node-node-name>`
+
+
+.. _prod-config-network-host:
+
+Network host
+------------
+
+By default, CrateDB binds to the loopback address (i.e., `localhost`_). It
+listens on port 4200-4299 for HTTP traffic and port 4300-4399 for node-to-node
+communication. Because CrateDB uses a port range, if one port is busy, it will
+automatically try the next port.
+
+When using :ref:`multiple hosts <multi_node_setup>`, nodes must bind to a
+non-loopback address. You can bind to a non-loopback address with the
+`network.host`_ setting.
+
+When CrateDB is bound to a non-loopback address, the :ref:`bootstrap checks
+<bootstrap-checks>` are enforced. This may require changes to your operation
+system configuration.
+
+.. CAUTION::
+
+      Never expose an unprotected CrateDB node to the public internet
+
+.. SEEALSO::
+
+    `Host settings`_
+
+
+.. _prod-config-heap:
+
+Heap
+----
+
+CrateDB is a Java application running on top of a Java Virtual Machine (JVM).
+The JVM uses a heap for memory allocations. For optimal performance, you must
+pay special attention to your heap configuration.
+
+.. SEEALSO::
+
+    :ref:`Optimizing memory performance <memory>`
+
+By default, CrateDB configures the JVM to dump out of memory exceptions to the
+file or directory specified by `CRATE_HEAP_DUMP_PATH`_. You must make sure
+there is enough disk space available for heap dumps at this location.
```

Comment:
Do we really want to talk about heap dumps at this point?
To me this would be another - really cool - how-to: debugging CrateDB.

Reply:
that's a cool idea for a new doc!! created an issue here: https://github.com/crate/crate-howtos/issues/209

however, I want to leave this in. the goal here is not to get into how to debug crate. it is to alert syadmins to the fact that they need to consider disk space requirements for this directory

leaving this in